winlogbeat/eventlog/eventlog.go

```diff
@@ -80,53 +82,64 @@ type Record struct {
 
 // ToMapStr returns a new MapStr containing the data from this Record.
 func (e Record) ToEvent() beat.Event {
-	m := common.MapStr{
-		"type":          e.API,
-		"log_name":      e.Channel,
-		"source_name":   e.Provider.Name,
-		"computer_name": e.Computer,
-		"record_number": strconv.FormatUint(e.RecordID, 10),
+	// Windows Log Specific data
+	win := common.MapStr{
+		"channel":       e.Channel,
 		"event_id":      e.EventIdentifier.ID,
+		"provider_name": e.Provider.Name,
+		"record_id":     e.RecordID,
+		"task":          e.Task,
+		"api":           e.API,
 	}
-
-	addOptional(m, "xml", e.XML)
-	addOptional(m, "provider_guid", e.Provider.GUID)
-	addOptional(m, "version", e.Version)
-	addOptional(m, "level", e.Level)
-	addOptional(m, "task", e.Task)
-	addOptional(m, "opcode", e.Opcode)
-	addOptional(m, "keywords", e.Keywords)
-	addOptional(m, "message", sys.RemoveWindowsLineEndings(e.Message))
-	addOptional(m, "message_error", e.RenderErr)
-
+	addOptional(win, "computer_name", e.Computer)
+	addOptional(win, "kernel_time", e.Execution.KernelTime)
+	addOptional(win, "keywords", e.Keywords)
+	addOptional(win, "opcode", e.Opcode)
+	addOptional(win, "processor_id", e.Execution.ProcessorID)
+	addOptional(win, "processor_time", e.Execution.ProcessorTime)
+	addOptional(win, "provider_guid", e.Provider.GUID)
+	addOptional(win, "session_id", e.Execution.SessionID)
+	addOptional(win, "task", e.Task)
+	addOptional(win, "user_time", e.Execution.UserTime)
+	addOptional(win, "version", e.Version)
+	addOptional(win, "message_error", e.RenderErr)
```

Comment:
I think this meets the description for `error.message`. And there is a `e.RenderErrorCode` field that could go into `error.code`.

Reply:
You make a good point.

I also noticed that RenderErrorCode was actually never passed along, we grab it, but eventlog.go never adds it to the event.

So I'll correct all that.

I've also noticed a problem for another time. The error message is actually populated either from parsing error (a Winlogbeat error message) or from the `ProcessingErrorData` accessor. This potentially clobbers userland data with pipeline data. But this is something we do across the board in all Beats, so unraveling that will take some time.

Reply:
I'd also be curious to know what the EventPayload blob is all about ([MSDN doesn't say much](https://docs.microsoft.com/en-us/windows/desktop/wes/eventschema-eventpayload-processingerrordatatype-element)). We're not touching that at all, we don't even have an accessor for it. Anyway, not touching that right now.